src/daft-dsl/src/expr/mod.rs

```diff
@@ -1755,6 +1793,21 @@ impl Expr {
             })?
             .data)
     }
+
+    pub fn bind(self: ExprRef, schema: &Schema) -> DaftResult<ExprRef> {
+        self.transform(|e| match e.as_ref() {
+            // TODO: remove ability to bind unresolved columns once we fix all tests
+            Self::Column(Column::Unresolved(UnresolvedColumn { name, .. }))
+            | Self::Column(Column::Resolved(ResolvedColumn::Basic(name))) => {
+                let index = schema.get_index(name)?;
+                let field = schema.get_field(name)?.clone();
+
+                Ok(Transformed::yes(bound_col(index, field)))
+            }
+            _ => Ok(Transformed::no(e)),
+        })
+        .map(|t| t.data)
+    }
```

Comment:
Nice! Once pushdowns use the python Expression, then I will remove the python-side binding and add this to PyExpr. I think that closes the loop for earlier conversations. https://github.com/Eventual-Inc/Daft/blob/main/daft/io/scan.py#L119-L126